bundles/tools.vitruv.change.atomic/src/tools/vitruv/change/atomic/EChangeUuidManager.xtend

```diff
@@ -0,0 +1,111 @@
+package tools.vitruv.change.atomic
+
+import org.eclipse.emf.ecore.EObject
+import tools.vitruv.change.atomic.eobject.CreateEObject
+import tools.vitruv.change.atomic.eobject.EObjectAddedEChange
+import tools.vitruv.change.atomic.eobject.EObjectExistenceEChange
+import tools.vitruv.change.atomic.eobject.EObjectSubtractedEChange
+import tools.vitruv.change.atomic.feature.FeatureEChange
+import tools.vitruv.change.atomic.uuid.UuidResolver
+
+import static com.google.common.base.Preconditions.checkArgument
+import static com.google.common.base.Preconditions.checkState
+
+/**
+ * Provides logic for initializing the UUIDs within changes. 
+ */
+class EChangeUuidManager {
+	val UuidResolver uuidResolver
+
+	/**
+	 * Initializes the manager with a {@link UuidResolver}.
+	 * 
+	 * @param uuidResolver -
+	 * 		the {@link UuidResolver} to use for UUID management
+	 */
+	new(UuidResolver uuidResolver) {
+		checkArgument(uuidResolver !== null, "uuid resolver must not be null")
+		this.uuidResolver = uuidResolver
+	}
+	
+	def static void setOrGenerateIds(Iterable<EChange> eChanges, UuidResolver uuidResolver) {
+		setOrGenerateIds(eChanges, uuidResolver, true)
+	}
+	
+	def static void setOrGenerateIds(Iterable<EChange> eChanges, UuidResolver uuidResolver, boolean endTransaction) {
+		val manager = new EChangeUuidManager(uuidResolver)
+		eChanges.forEach [ eChange |
+			manager.setOrGenerateIds(eChange)
+		]
+		if (endTransaction) {
+			uuidResolver.endTransaction
+		}
+	}
+
+	def void setOrGenerateIds(EChange eChange) {
+		switch eChange {
+			CreateEObject<?>:
+				setOrGenerateCreatedEObjectUuid(eChange)
+			EObjectExistenceEChange<?>:
+				setAffectedEObjectUuid(eChange)
+			FeatureEChange<?,?>:
+				setAffectedEObjectUuid(eChange)
+		}
+		switch eChange {
+			EObjectSubtractedEChange<?>:
+				setOldValueUuid(eChange)
+		}
+		switch eChange {
+			EObjectAddedEChange<?>:
+				setOrGenerateNewValueUuid(eChange)
+		}
+	}
+
+	private def String getUuid(EObject object) {
+		val id = uuidResolver.getUuid(object)
+		checkState(id !== null, "uuid must not be null")
+		return id
+	}
+	
+	private def String getOrGenerateUuid(EObject object) {
+		try {
+			getUuid(object)
+		}
+		catch (IllegalStateException e) {
+			uuidResolver.registerEObject(object)
```

Comment:
The UuidResolver interface has a `hasUuid` method so that there is no need to catch this exception.

Reply:
Resolved in https://github.com/vitruv-tools/Vitruv-Change/pull/47/commits/fd0a8b4218ba42cd94c8eb8ae2567ffcc4bb5526